### DISCUSSION TITLE
Be able to open the Wallets UI Modal even when connected

### DISCUSSION THREAD

**Discussion by FiboApe** [rainbow-me/rainbowkit#2042] [Feedback]
Hey Rainbow team, Fibo here from the Balmy team (Ex-Mean Finance)!

Love rainbowkit and we use it on our dApp. In Balmy we allow bundling wallets for one user where they can connect different wallets and we track them all for them, pretty standard. The only issue is for our users to either chain their connected wallet, or connect a new one, we need to disconnect the current wallet and then open the wallet modal, this means that if a user actually then decides to not connect any new wallet, their previous one remains disconnected since we have to do it manually.

Would love to be able to open the Wallets modal without having to disconnect the users wallet through wagmi first. 

Just to give an example what we do today is this [this](https://github.com/Balmy-protocol/dca-fe/blob/staging/apps/root/src/hooks/useOpenConnectModal.ts) it has a bunch of css there since we kinda modify a bit the structure of the wallet modal but the important part is that we have to disconnect to open the modal again.

LMK what you think, I'm open to create a PR adding this feature :)

**Comment by 0xmkh** [comment]
@FiboApe Yeah for now the only way to open the modal is if you are disconnected. We use Wagmi on top of RainbowKit to detect that. 

Do you think it'll be possible to create your own connector where you can manage the connected / disconnected state ? Here is an example of [createConnector](https://wagmi.sh/core/api/createConnector#createconnector) function from wagmi. Let me know if this is helpful and we can discuss on how we can approach this on our end 👍

**Comment by tonypeng** [comment]
@magiziz any chance we could get an option to force the modal to open if a wallet is already connected?

working on an app with a similar use case where a user can link multiple wallets to their account as authentication methods. As of now, if the user has previously connected Wallet 1, and currently has Wallet 2 active (e.g. as the currently selected wallet in Metamask), and presses `Connect`, the modal will refuse to open unless we first disconnect Wallet 1.

this results in needing to do hacky code like so:
```
  React.useEffect(() => {
    // hack: we have to disconnect any connected wallet to be able to show the connect modal
    if (connectedWalletAddress) {
      if (connectInProgressRef.current) {
        signMessageAsync({
          message: /* ... */,
        })
          .then(async (signature) => {
            // ...
          })
          .catch(() => /* ... */)
          .finally(() => {
            disconnect();
          });
      } else {
        disconnect();
      }
    }
  }, [connectedWalletAddress]);
```

Also, as a side note, is it possible for `openConnectModal` to return a Promise resolving when a wallet has been completed? As of now we're setting `connectInProgressRef.current` on button click to prevent the above hook from asking for a signature on initial page load, if a wallet is already connected. If we could instead await a Promise from connecting a wallet it would be cleaner and more reliable to prompt for a signature right after a wallet gets connected.
